pkg/controller/certificates/trigger/trigger_controller_test.go

```diff
@@ -97,412 +77,266 @@ func Test_controller_ProcessItem(t *testing.T) {
-				{
-					ObjectMeta: metav1.ObjectMeta{
-						Namespace: "testns",
-						Name:      "test",
-						Annotations: map[string]string{
-							cmapi.CertificateRequestRevisionAnnotationKey: "3",
-						},
-						OwnerReferences: []metav1.OwnerReference{
-							*metav1.NewControllerRef(&cmapi.Certificate{ObjectMeta: metav1.ObjectMeta{Namespace: "testns", Name: "test"}}, cmapi.SchemeGroupVersion.WithKind("Certificate")),
-						},
-					},
-				},
+		"should call shouldReissue with the correct cert, secret and current CR": {
+			existingCertificate: gen.Certificate("cert-1", gen.SetCertificateNamespace("testns"),
+				gen.SetCertificateSecretName("secret-1"),
+				gen.SetCertificateGeneration(42),
+				gen.SetCertificateUID("cert-1-uid"),
+				gen.SetCertificateRevision(2),
+			),
+			wantDataForCertificateCalled: true,
+			mockDataForCertificateReturn: policies.Input{
+				Secret: gen.Secret("secret-1", gen.SetSecretNamespace("testns")),
+				CurrentRevisionRequest: gen.CertificateRequest("cr-1", gen.SetCertificateRequestNamespace("testns"),
+					gen.AddCertificateRequestOwnerReferences(gen.CertificateRef("cert-1", "cert-1-uid")),
+					gen.SetCertificateRequestAnnotations(map[string]string{"cert-manager.io/certificate-revision": "2"}),
+				),
 			},
-			chainShouldEvaluate: true,
-			policyFuncs: []policyFuncBuilder{
-				// Add a policy function that ensures only the input's 'certificate'
-				// field is set.
-				func(t *testing.T) policies.Func {
-					return func(input policies.Input) (string, string, bool) {
-						if input.Certificate == nil {
-							t.Error("expected policy data 'Certificate' field to be set but it was not")
-						}
-						if input.Secret == nil {
-							t.Errorf("expected policy data 'Secret' field to be set but it was not")
-						}
-						if input.CurrentRevisionRequest == nil {
-							t.Errorf("expected policy data 'CurrentRevisionRequest' field to be set but it was not")
-						}
-						return "", "", false
+			wantShouldReissueCalled: true,
+			mockShouldReissue: func(t *testing.T) policies.Func {
+				return func(gotInput policies.Input) (string, string, bool) {
+					expectInput := policies.Input{
+						Certificate: gen.Certificate("cert-1", gen.SetCertificateNamespace("testns"),
+							gen.SetCertificateSecretName("secret-1"),
+							gen.SetCertificateGeneration(42),
+							gen.SetCertificateUID("cert-1-uid"),
+							gen.SetCertificateRevision(2),
+						),
+						CurrentRevisionRequest: gen.CertificateRequest("cr-1", gen.SetCertificateRequestNamespace("testns"),
+							gen.AddCertificateRequestOwnerReferences(gen.CertificateRef("cert-1", "cert-1-uid")),
+							gen.SetCertificateRequestAnnotations(map[string]string{"cert-manager.io/certificate-revision": "2"}),
+						),
+						Secret: gen.Secret("secret-1", gen.SetSecretNamespace("testns")),
 					}
-				},
-			},
-		},
-		"error if multiple owned CertificateRequest resources exist and have the same revision": {
```

Comment:
This test has vanished, do we still have the same coverage?

Reply:
Thanks for bringing this up, I had forgotten to explain where this test case has been moved to 😅

I removed this test case since it is now in [gatherer_test.go](https://github.com/jetstack/cert-manager/blob/3a367927dcec94fa55aa2e881db5b71cf79889fa/pkg/controller/certificates/trigger/policies/gatherer_test.go#L162-L186)